src/main/java/net/shadowmage/ancientwarfare/structure/item/ItemBlockPosts.java

```diff
@@ -0,0 +1,52 @@
+package net.shadowmage.ancientwarfare.structure.item;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.Blocks;
+import net.minecraft.util.EnumActionResult;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.shadowmage.ancientwarfare.core.item.ItemBlockBase;
+import net.shadowmage.ancientwarfare.structure.util.MultiBlockHelper;
+
+public class ItemBlockPosts extends ItemBlockBase {
+	public ItemBlockPosts(Block block) {
+		super(block);
+	}
+
+	@Override
+	public EnumActionResult onItemUse(EntityPlayer player, World world, BlockPos pos, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ) {
+		return MultiBlockHelper.onMultiBlockItemUse(this, player, world, pos, hand, facing, hitX, hitY, hitZ, this::mayPlace);
+	}
+
+	private boolean mayPlace(World world, BlockPos pos, EnumFacing sidePlacedOn, EntityPlayer placer) {
+		return canPlaceVertical(world, pos, sidePlacedOn);
+	}
+
+	public static boolean canPlaceVertical(World world, BlockPos pos, EnumFacing sidePlacedOn) {
+		for (int offset = 0; offset < 4; offset++) {
+			if (!mayPlaceAt(world, pos.offset(EnumFacing.UP, offset), sidePlacedOn, offset == 0)) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	private static boolean mayPlaceAt(World world, BlockPos pos, EnumFacing sidePlacedOn, boolean checkSide) {
+		IBlockState state = world.getBlockState(pos);
```

Comment:
The logic here should check for the full size of the multiblock so that it doesn't replace non replaceable blocks (check 4 blocks in a column starting from the one where player places it)

Reply:
Sorry just realized you have that right above, my bad!